handlers/_swagger.js

```diff
@@ -419,40 +423,52 @@ function validateRequestParameters(req, data, logger, callback) {
 
         } else if (parm.in === 'body') {
             var result = swaggerUtil.validateJSONType(parm.schema, req.body);
-
-            if (!result.valid) {
+            var polymorphicValidationErrors = [];
+            if (polymorphicValidation) {
+                polymorphicValidationErrors = swaggerUtil.validateIndividualObjects(swaggerDoc, parm['x-bos-generated-disc-map'], req.body);
+            }
+            if (!result.valid || polymorphicValidationErrors.length != 0) {
                 var error = new VError('Error validating request body');
                 error.name = 'ValidationError';
-                error.subErrors = result.errors;
+                error.subErrors = result.errors.concat(polymorphicValidationErrors);
                 return callback(error);
             }
         }
     }
     return callback();
 }
 
-function validateResponseModels(res, body, data, logger) {
+function validateResponseModels(res, body, data, logger, swaggerDoc) {
     if (!(res.statusCode >= 200 && res.statusCode < 300 || res.statusCode >= 400 && res.statusCode < 500)) {
         //the statusCode for the response isn't in the range that we'd expect to be documented in the swagger
         //i.e.: 200-299 (success) or 400-499 (request error)
         return; 
     }
 
     var schemaPath = 'responses.%s.schema',
+        mapPath = 'responses.%s.x-bos-generated-disc-map',
         codeSchema = util.format(schemaPath, res.statusCode),
-        defaultSchema = util.format(schemaPath, 'default');
+        defaultSchema = util.format(schemaPath, 'default'),
+        mapSchema = util.format(mapPath, res.statusCode),
+        defaultMapSchema = util.format(mapPath, 'default');
     var modelSchema;
+    var responseModelMap;
     if (_.has(data, codeSchema)) {
         modelSchema = _.get(data, codeSchema);
+        responseModelMap = _.get(data, mapSchema);
     } else if (_.has(data, defaultSchema)) {
         modelSchema = _.get(data, defaultSchema);
+        responseModelMap = _.get(data, defaultMapSchema);
     } else {
         return _createValidationError('No response schema defined for %s %s with status code %s');
     }
-    
     var result = swaggerUtil.validateJSONType(modelSchema, body);
-    if (!result.valid) {
-        return _createValidationError('Error validating response body for %s %s with status code %s', result.errors);
+    var polymorphicValidationErrors = [];
```

Comment:
can the whole basic validation + polymorphic validation be extracted to a common function?
(it looks like the code is largely the same)